### PR TITLE
Health Check Code and Nuget Version Updated

### PR DESCRIPTION
**Pull request by v-shaaal**
## Description
- Updated health check code after the shared piece of code was moved to healthcare shared components solution.
- Updated nuget packages to point to the latest version of healthcare shared components solution.
- Fixed an existing bug which was caused because of an update in the shared components code. Corresponding change was however not made in FHIR. This pull request fixes that issue.

## Testing
- All existing tests are passing.
- Health check tests are passing as well.

### CODE REVIEW

File: src/Microsoft.Health.Fhir.SqlServer/Features/Storage/SqlServerFhirModel.cs [hunk 1]
```diff
@@ -330,7 +329,7 @@ private void ThrowIfNotStarted()
         {
             if (!_started)
             {
-                _logger.LogError($"The {nameof(SqlServerModelInitializer)} instance has not been initialized.");
+                _logger.LogError($"The {nameof(SqlServerFhirModel)} instance has not been initialized.");
```
Comment (brandonpollett) [inline src/Microsoft.Health.Fhir.SqlServer/Features/Storage/SqlServerFhirModel.cs:332]: FYI: I think this is also being done in #1142

**Comment by v-shaaal** [comment]
> Where are we calling the `app.UseHealthChecksExtension(new PathString(KnownRoutes.HealthCheck));`?

I missed that change. Will be publishing a new iteration shortly with this change included.

**Comment by v-shaaal** [comment]
> > Where are we calling the `app.UseHealthChecksExtension(new PathString(KnownRoutes.HealthCheck));`?
> 
> I missed that change. Will be publishing a new iteration shortly with this change included.

I have published an iteration with the respective change.

File: src/Microsoft.Health.Fhir.SqlServer/Features/Storage/SqlServerFhirModel.cs (hunk 1 above)
Comment (rotodd) [reply src/Microsoft.Health.Fhir.SqlServer/Features/Storage/SqlServerFhirModel.cs:332]: Thanks for pointing this out. I will close #1142

**Pull request merged**